etcdctl/util/normalizer.go

```diff
@@ -0,0 +1,47 @@
+// Copyright 2024 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import "strings"
+
+const Indentation = "  "
+
+// Examples normalizes a command's examples to follow the conventions.
+func Examples(s string) string {
+	if len(s) == 0 {
+		return s
+	}
+	return normalizer{s}.trim().indent().string
+}
+
+type normalizer struct {
+	string
+}
+
+func (s normalizer) trim() normalizer {
+	s.string = strings.TrimSpace(s.string)
+	return s
+}
+
+func (s normalizer) indent() normalizer {
+	indentedLines := []string{}
+	for _, line := range strings.Split(s.string, "\n") {
+		trimmed := strings.TrimSpace(line)
+		indented := Indentation + trimmed
+		indentedLines = append(indentedLines, indented)
+	}
+	s.string = strings.Join(indentedLines, "\n")
+	return s
+}
```

Comment:
I'm not entirely sold on introducing this struct. However, I see why you did it. As I was reviewing, some comments/observations came to my mind:

1. I usually see the receiver variable named after its struct. In this case, I'd suggest using `n` rather than `s` (https://google.github.io/styleguide/go/decisions#receiver-names).
2. I think that modifying the held string's value is not transparent.
3. I don't know if this struct will be reused or if it is going to be used just for this context.

Because of these reasons, I wonder if it would be easier (and simpler) to define a single function that does the normalization without adding a struct that does it.

Reply:
Thanks, you are right i used pretty old code structure here, i will update them. 

For number 3 we don't need it to be structure now, but in future this function can evolve fast and add more thing to it if we want to extend normalizer function to other parts when we want to use exports, add more attributes to the `etcdctl` cobra that need some kind of normalizer.
And it is simpler/clearer to me/personally to use struct than using an function because i can see the function's parent and i can extend it's functionality easier. But if you think it is not needed i can replace it with a simple function right now!

Reply:
Can you please define this phrase? 
> I think that modifying the held string's value is not transparent.

Reply:
It could be biased or a personal preference, but in my opinion, returning a modified value rather than altering the property from the structure is less error-prone. I think the pattern of returning the struct to then run another function on it is more common in NodeJS than in Go.

So, in a nutshell, I think an implementation like the following would be cleaner:

```golang
func (s normalizer) indent() normalizer {
	...
	return normalizer{strings.Join(indentedLines, "\n")}
}
```

But, again, the pattern of returning the `normalizer` so it can be chained with other functions is something that I haven't seen in Go a lot. So, this could be cleaner:

```golang
func (s normalizer) indent() string {
	...
	return strings.Join(indentedLines, "\n")
}
```

That's why I question the use of this structure, as it seems like it could be simplified to a simple, unexported function.

With the caveats, I may be missing the bigger picture if you think this could be reused in later implementations as part of your bigger effort to have more examples across the CLI tools.

Reply:
Thanks for explaining i see what you mean, TBH currently the implementation is just about using it in the function here, with those concerns i can do two things
1. replacing structure with a simple function and for later if we needed we can do a structure
2. altering the current behavior from returning structure to returning string to be more go style

Your'e call i'm ok either way.

Reply:
@ahrtr, do you have an opinion on this discussion?

Reply:
Overall not a big deal to me. Since it's just internal implementation, we can update it later when needed.

The `normalizer{s}.trim().indent().string` is a good pattern to me. It reflects the ideas of functional programming and also (although compromised) object-oriented programming. 

But on the other hand, the implementation is pretty simple, so we can also just get it included in one function. In other words, no need to add `normalizer` struct at all.

Again, as mentioned above, it isn't a big deal. This is solely for the purpose of a purely technical discussion. 

Let's just keep the current implementation as it's if there isn't no strong objection.